spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractMessageHandler.java

```diff
@@ -144,9 +144,9 @@ public void handleMessage(Message<?> message) {
 		MetricsContext start = null;
 		boolean countsEnabled = this.countsEnabled;
 		AbstractMessageHandlerMetrics handlerMetrics = this.handlerMetrics;
-		Sample sample = null;
-		if (countsEnabled && this.meterRegistry != null) {
-			sample = Timer.start(this.meterRegistry);
+		Object sample = null;
+		if (countsEnabled && this.metricsCaptor != null) {
+			sample = this.metricsCaptor.start();
```

Comment:
This API is somehow is not consistent with the `Timer.start(this.meterRegistry)`. 
Maybe we can do something similar as well like `TimerFacade.start(this.metricsCaptor)`?

That will help us with narrowing generic arg scope as well